src/common/offline_transformations/src/pruning/propagate_masks.cpp

```diff
@@ -63,11 +67,24 @@ class ngraph::pass::mask_propagation::MatMul : public MatcherPass {
             auto a_mask = getMask(m_a);
             auto b_mask = getMask(m_b);
 
-            if (!a_mask || !b_mask) {
+            if (!a_mask && !b_mask) {
```

Comment:
this check looks unnecessary, is it only for debug msg?

Reply:
This check is essential, following configuration could propagate pruning:
![image](https://user-images.githubusercontent.com/74656388/181236176-84fe504e-f4f9-4eac-867d-3ac1f91439db.png)

Reply:
as you can see from your table "can_propagate" depends on b_mask and doesn't depends on a_mask :)
and you already have this check in the code below

Reply:
Ok, you are right, sorry for my dump 👍 
I think it's ok to keep this, this info is helpful in logs

Reply:
ok!